File: Leetcode/2379. Minimum Recolors to Get K Consecutive Black Blocks.py
class Solution:
    def minimumRecolors(self, blocks: str, k: int) -> int:
        i=0
        j=k-1
        n=len(blocks)
        count=blocks[:k].count('W')
        res=count
        # print(j,n-1)
        while j<n-1:
            j+=1
            if blocks[j]=='W':
                count+=1
            if blocks[i]=='W':
                count-=1
            i+=1
            # print(count)
            res=min(count,res)
        return res
        

File: Leetcode/2379. Minimum Recolors to Get K Consecutive Black Blocks.cpp
class Solution {
public:
    int minimumRecolors(string blocks, int k) {
        int i=0,j=k-1,count=0,n=blocks.size();
        for(int p=0;p<k;p++){
            if(blocks[p]=='W')  count++;
        }
        int res=count;
        while(j<n-1){
            j++;
            if(blocks[j]=='W')  count++;
            if(blocks[i]=='W')  count--;
            i++;
            res=min(res,count);
        }
        return res;
    }
};
